bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -266,6 +269,18 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
                     convertibleTypes.add(matchingType);
                 }
                 break;
+            case TypeTags.FINITE_TYPE_TAG:
+                for (Object valueSpaceItem : ((BFiniteType) targetType).valueSpace) {
+                    Type inputValueType = TypeChecker.getType(inputValue);
+                    Type valueSpaceItemType = TypeChecker.getType(valueSpaceItem);
+                    if (inputValue == valueSpaceItem && inputValueType.getTag() == valueSpaceItemType.getTag()) {
```

Comment:
Can you please share an example for when the value is the same but the type tag is different?

Reply:
There should be no possible cases, just added for a safe check but it looks like redundant. Removed with a later commit.